### ISSUE TITLE
Option to replace dups with symbolic links

### ISSUE THREAD

**Issue by 1024mb** [ermig1979/AntiDupl#75]
Can you add an option to replace duplicates with symlinks keeping the original filenames? It shouldn't matter if there is a dup group with a jpg and a png, the program should make a symlink pointing to the png file (or the jpg, ofc it all depends on which one we select to replace) but in its filename it should contain the original extension and name.

So if i.e. we have Photo 1.jpg and Photo34.png then AntiDupl should replace Photo 1.jpg with a symlink pointing to Photo34.png but keeping its original name and extension (Photo 1.jpg).

**Comment by malcalevak** [comment]
While I suppose AntiDupl finds visual matches across formats, it feels wrong to me to create a symlinks that would have different extensions. That said, your request does have me wanting to see if there are any requests for hardlinks.

**Comment by 1024mb** [comment]
> 
> 
> While I suppose AntiDupl finds visual matches across formats, it feels wrong to me to create a symlinks that would have different extensions. That said, your request does have me wanting to see if there are any requests for hardlinks.

Of course, that's valid and that's why it (as everything) should be optional.
Also if AntiDupl finds two or more dups with different format, it is almost always the case that the one with the larger size is the one with the better quality so having a JPEG with 50kb and a PNG with 4mb of the same pic one would obviously want to keep the PNG and delete the JPEG file and, because you have them in different directories (because reasons), you would want to keep the file and point it to the other one so you don't loose the structure. Is like having two folders one says "Wedding" and the other "Photos with XXX" and in the "Wedding" folder you have the lossless file your friend the photographer send you and in the "Photos with XXX" folder you have the JPG you downloaded from Twitter or Facebook and they both are the same photo, if Antidupl would detect them everybody would want to keep the lossless file and also keep the other one but pointing to the file with the better quality and/or resolution. And like that there are a lot of other cases were this would be useful.
